### PR TITLE
vagrant: Install Go 1.4.2 instead of default Ubuntu Go.

### PR DESCRIPTION
**Pull request by ryan-endacott**
Go 1.4.2 is the latest version, so I figured we should use it.  One of the packages we're using also didn't work with the default Ubuntu Go version (1.2 IIRC).  

This also fixes a bug in the vagrant installation process caused by filename case sensitivity.

### CODE REVIEW

File: Vagrant/scripts/goDeps.sh [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export GOPATH=/home/vagrant/go;
+export PATH=$PATH:/usr/local/go/bin;
```
Comment (ryan-endacott) [inline Vagrant/scripts/goDeps.sh]: Without this line, this script couldn't find the Go command.

File: Vagrant/scripts/bootstrap.sh [hunk 2]
```diff
@@ -1,7 +1,11 @@
 sudo apt-get -y install git;
 sudo apt-get -y install mercurial;
 
-sudo apt-get -y install golang;
+# Install Go 1.4.2
+wget --quiet https://storage.googleapis.com/golang/go1.4.2.linux-amd64.tar.gz
+tar -C /usr/local -xzf go1.4.2.linux-amd64.tar.gz
+echo "export PATH=$PATH:/usr/local/go/bin" >> /home/vagrant/.profile;
+export PATH=$PATH:/usr/local/go/bin;
```
Comment (Felttrip) [inline Vagrant/scripts/bootstrap.sh]: Looks good,  good call on making sure we get the most up to date go.

File: Vagrant/scripts/goDeps.sh (hunk 1 above)
Comment (Felttrip) [inline Vagrant/scripts/goDeps.sh]: I bet the apt-get handled that for us earlier.

**Comment by Felttrip** [comment]
@ryan-endacott reviewed and looks good. We should talk about getting a standard labeling system on GH

**Comment by ryan-endacott** [comment]
Thanks for the review!

**Comment by ryan-endacott** [comment]
Note:  In the future, I'll merge into master without a merge commit.  Didn't realize GitHub would make one if the branch was already rebased on master.

**Pull request merged**